planner/core/rule_aggregation_push_down.go

```diff
@@ -316,7 +317,7 @@ func (a *aggregationOptimizer) optimize(p LogicalPlan) (LogicalPlan, error) {
 }
 
 // aggPushDown tries to push down aggregate functions to join paths.
-func (a *aggregationOptimizer) aggPushDown(p LogicalPlan) LogicalPlan {
+func (a *aggregationPushDownSolver) aggPushDown(p LogicalPlan) LogicalPlan {
 	if agg, ok := p.(*LogicalAggregation); ok {
 		proj := a.tryToEliminateAggregation(agg)
```

Comment:
can we remove this?

Reply:
can we make the aggregate pruning happens after the aggregate pushdown?

Reply:
We cannot. `aggregate prune` is also used for https://github.com/pingcap/tidb/pull/7531. So it cannot be operated too late.

Reply:
sadness.